forest/src/main.rs

```diff
@@ -8,8 +8,35 @@ use async_std::task;
 use forest_libp2p::{get_keypair, Libp2pService};
 use libp2p::identity::{ed25519, Keypair};
 use log::{info, trace};
+use std::cell::RefCell;
+use std::process;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
 use utils::write_to_file;
 
+// Blocks current thread until ctrl-c is received
+fn block_until_sigint() {
+    let (ctrlc_send, ctrlc_oneshot) = futures::channel::oneshot::channel();
+    let ctrlc_send_c = RefCell::new(Some(ctrlc_send));
+
+    let running = Arc::new(AtomicUsize::new(0));
+    ctrlc::set_handler(move || {
+        let prev = running.fetch_add(1, Ordering::SeqCst);
+        if prev == 0 {
+            println!("Got interrupt, shutting down...");
```

Comment:
nit: "Got interrupt**ed**, shutting down..."

Reply:
I think the noun version makes more sense in this context, anyone else have a preference? Idrc happy to change

Reply:
I'm okay with either. They're both clear enough. Maybe "Shutdown signal received... Shutting down" might be better? Again, indifferent though

Reply:
haha to avoid going down this rabbit hole I'm just going to pull in as is, I'll approve any change if anyone cares that much :)